keps/sig-network/2091-admin-network-policy/README.md

```diff
@@ -402,182 +393,262 @@ The following new `AdminNetworkPolicy` API will be added to the `policy.networki
   Spec     AdminNetworkPolicySpec
-  // Status to be reported by the CNI, this is not standardized in alpha and 
-  // CNIs should report what they see fit in relation to their AdminNetworkPolicy
-  // implementation
+
+  // ANPStatus is the status to be reported by the implementation, this is not 
+  // standardized in alpha and consumers should report what they see fit in 
+  // relation to their AdminNetworkPolicy implementation
   // +optional 
-  Status   []metav1.Condition
+  Status  AdminNetworkPolicyStatus
 }
 
-type AdminNetworkPolicySpec struct {
-  // Int value bound to 0 - 999, with 0 signaling the special "baseline" priority
-  Priority      Int 
-  // Allows for multiple AdminNetworkPolicy rules at a single priority level
-  // If empty no rules are applied
-  // A total of 100 total rules will be allowed per each network policy instance, 
-  // this rule count will be calculated as the total summation of the 
-  // AdminNetworkPolicyRules, AdminNetworkPolicyIngressRules, and AdminNetworkPolicyEgressRules
-  // in a single AdminNetworkPolicy Instance. 
-  PolicyRules   []AdminNetworkPolicyRule
+type AdminNetworkPolicyStatus struct { 
+  conditions   []metav1.Condition  
 }
 
-type AdminNetworkPolicyRule struct {
-  // No implicit isolation of AppliedTo Pods/Namespaces. 
-  // If no rules are a defined no behavior will be observed 
-  AppliedTo   AppliedTo
+// AdminNetworkPolicySpec provides the specification of AdminNetworkPolicy
+type AdminNetworkPolicySpec struct {
+  // Priority is an int32 value bound to 0 - 1000, the lowest positive priority, 
+  // "1" corresponds to the lowest importance, while higher priorities have 
+  // higher importance. An ANP with a priority of "0" will be evaluated after all 
+  // positive priority AdminNetworkPolicies and standard NetworkPolicies.
+  // The Priority for an ANP must be set 
+  Priority    *int32
+
+  // Subject defines the objects to which this AdminNetworkPolicy applies.  
+  Subject     AdminNetworkPolicySubject
+
+  // List of Ingress rules to be applied to the selected objects.
+  // A total of 100 rules will be allowed per each network policy instance, 
+  // this rule count will be calculated as the total summation of the 
+  // Ingress and Egress rules in a single AdminNetworkPolicy Instance. 
   Ingress     []AdminNetworkPolicyIngressRule
+
+  // List of Egress rules to be applied to the selected objects.
+  // A total of 100 rules will be allowed per each network policy instance, 
+  // this rule count will be calculated as the total summation of the 
+  // Ingress and Egress rules in a single AdminNetworkPolicy Instance. 
   Egress      []AdminNetworkPolicyEgressRule
 }
 
-// Exactly one of the selector pointers should be set in a given AppliedTo
-type AppliedTo struct {
+// AdminNetworkPolicySubject defines what objects the policy selects. 
+// Exactly one of the selector pointers should be set.  
+type AdminNetworkPolicySubject struct {
```

Comment:
Why not make this have the same form as NetworkPolicyPeer - just `NamespaceSelector` and `PodSelector`? The validation rules would be different but at least the semantics would be less surprising.

Reply:
Oh, nevermind, I get it. This gets rid of the awkward "you can set ipBlock OR Namespace OR Namespace and Pod" wart in NP.

Reply:
It also allows us to add selectors in the future without breaking implementations.... Since now the non-existence of a `NamespaceSelector` No Longer means "all namespaces"

Reply:
For the sake of KEP deadline, we can defer naming to the API review

Reply:
++

Reply:
Done..... For now :)